Add updated Jupyter Notebooks for LLM testing

rhysmah opened the pull request:
## Summary
This is a Jupyter Notebook with some basic code to load a Llama3 model; create some agents, tasks, and a crew; read from a file containing functions; then write the updated functions to another file.

This pipeline is not yet finished.

## Testing

** I tried running this without Ollama running. It did not work. For those on Windows and Linux, please try running this twice -- once with Ollama running, once without. **

1. Checkout the `rhys/research_update` branch
2. Open the integrated terminal in VSCode
3. Run `poetry run jupyter notebook`
4. Navigate to `rhys_research > LLM_Test`
5. At the top of the screen, click "Kernel", then "Restart Kernel and Run All Cells..."
6. Once processing ends, check the `test_outputs` folder and see if a file was written.

carcango reviewed (approved):
Works great! Nice job

tgrigoruk reviewed (changes requested):
I haven't looked much into the content, that's for the selected reviewers. I did notice a few issues though:

Needs addressing:
- files/folders starting with a period are "hidden" and generally shouldn't be pushed (except for certain config files)
	-  e.g. you should add `.pynb_checkpoints/` to the `.gitignore` as those files are redundant clutter
- you should avoid naming things after yourself (folder, section in pyproject.toml) - just call something what it is, and if someone needs to know who made the change they can look at the git history
- there is a file that is literally Untitled

Things to consider:
- your commit messages are formatted like branch names, which is very confusing and weird. Please refer to the **Commit Naming Convention** section in the [style guide](https://dev.to/varbsan/a-simplified-convention-for-naming-branches-and-commits-in-git-il4) I shared earlier
	- be mindful if which kind of slash you use - for paths (which branches are like) use a forward slash `/` 
	- don't bother changes these now as they should all be squashed before merging
- A PR title should be very similar to the commit message of the squashed merge, and should be more descriptive, e.g. something like "Add jupyter notebooks with updated research" (on squash it gets renamed `feat: add ...`)
- Create your PR as a Draft first, that way it runs through all the checks and you can see if changes need to be made (like fix errors, or fix conflicts with main) _before_ requesting others review your work

tgrigoruk commented on main.py:
This file has been moved (and significantly altered) - probably best to delete it.

tgrigoruk commented on Untitled.ipynb:
Aside from obviously missing a name, this file should be moved elsewhere. The project root directory (in our case) should not contain any source code.

tgrigoruk commented on python-package/src/package/ai/tasks.py:
Tasks have been renamed here - references to them in `crew.py` may need to be updated accordingly.

rhysmah commented:
There was discussion about whether Ollama needed to be running for Jupyter Notebook code to work. According to Mike, (Linux), Ollama was started automatically, whereas for me (Mac), it had to be started manually. In other words, Ollama **is** required for the code in Jupyter Notebooks to run properly.

rhysmah commented:
The files within `python-package/src/package/ai` are outside the purview of this PR. However, to Terence's point, the tasks defined in `crew.py` are not the same as the tasks in `tasks.py`, so I've deleted the tasks within `crew.py` and added a comment. These files may change substantially, so I won't make any more updates.

rhysmah commented:
@tgrigoruk, thank you for the in-depth comments. I believe I've addressed most or all of your points (and have used the got style guide you recommended. Let me know if you have any additional comments -- always open to learning how I can improve.

tgrigoruk commented on pyproject.toml:
```diff
@@ -15,11 +15,15 @@ crewai = ">=0.28.8,<0.31.0"
 pre-commit = "^3.7.0"
 ruff = "^0.4.3"
 
+[tool.ruff]
+exclude = ["llm_testing/test_inputs/test_functions_raw.py"]
+
```
All settings for `ruff` should be in `.ruff.toml` - that way it's all in one place (you would put it here if that file didn't exist). See the `exclude` section.
It would also be a good idea to put _all_ test code files in one place, unless there's good reason to do otherwise.

rhysmah commented:
## Updated Testing Instructions

1. Check out the `rhys/research_update` branch
2. Open the integrated terminal
3. Run `jupyter notebook`; a browser will automatically open
4. Navigate to `llm_testing/LLM_Test` 
5. At the top of the screen, click "Kernel" > "Restart Kernel and Run All Cells..."

tgrigoruk commented on pyproject.toml:
```diff
@@ -15,6 +15,7 @@ crewai = ">=0.28.8,<0.31.0"
 langchain-anthropic = "^0.1.11"
 langchain-groq = "^0.1.3"
 ollama = "^0.2.0"
+jupyter = "^1.0.0"
```
`jupyter` is not required by the package to run - it is to be used by developers - and should be included in `...dev.dependencies` below (to prevent it from being included in the packages shipped with the extension)
